### PR TITLE
🌱 [wip] test: clusterctl upgrade test on v1.31.0

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by chrischdi** [comment]
Note: expecting it to fail!

**Comment by sbueringer** [comment]
Perfect:
> action failed after 10 attempts: failed to patch provider object: CustomResourceDefinition.apiextensions.k8s.io "clusterclasses.cluster.x-k8s.io" is invalid: spec.conversion.webhookClientConfig.caBundle: Invalid value: []byte{0xa}: unable to load root certificates: unable to parse bytes as PEM block

@chrischdi WDYT about noting in our tracking issue that we should bump this version as well going forward?

**Comment by sbueringer** [comment]
Pushed the fix from https://github.com/kubernetes-sigs/cluster-api/pull/10972 to verify it works with the fix

**Comment by sbueringer** [comment]
/test pull-cluster-api-e2e-main

Now expecting it to work

File: test/e2e/clusterctl_upgrade_test.go [hunk 1]
```diff
@@ -292,8 +292,8 @@ var _ = Describe("When testing clusterctl upgrades (v1.7=>current)", func() {
 			InitWithBinary:            fmt.Sprintf(clusterctlDownloadURL, stableRelease),
 			InitWithProvidersContract: "v1beta1",
 			//  Note: Both InitWithKubernetesVersion and WorkloadKubernetesVersion should be the highest mgmt cluster version supported by the source Cluster API version.
-			InitWithKubernetesVersion:   "v1.30.0",
-			WorkloadKubernetesVersion:   "v1.30.0",
+			InitWithKubernetesVersion:   "v1.31.0-rc.0",
+			WorkloadKubernetesVersion:   "v1.31.0-rc.0",
```
Comment (sbueringer) [inline test/e2e/clusterctl_upgrade_test.go:296]: I think bumping the wl cluster version was too much. CAPI 1.7 doesn't support it

**Comment by sbueringer** [comment]
Going to close this one, tests completed

/close

**Pull request closed without merging**